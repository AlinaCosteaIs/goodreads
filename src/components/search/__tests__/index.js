import React from 'react'
import Search from '../index'
import Button from '../components/button'
import Input from '../components/input'
import TestRenderer from 'react-test-renderer'

let props = {}
const mockFn = jest.fn()
const event = { target: { value: 'search term' } }

beforeEach(() => {
  console.log('Running first test')
  props = {
    search: mockFn,
  }
})

test('Search component is rendered ok', () => {
  const component = TestRenderer.create(<Search {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('Button click event is handled', () => {
  const testRenderer = TestRenderer.create(<Search {...props} />)
  const testInstance = testRenderer.root
  const button = testInstance.findByType(Button)
  button.props.onClick()
  expect(testInstance.props.search).toHaveBeenCalled()
})

test('Input handles change events', () => {
  const testRenderer = TestRenderer.create(<Search {...props} />)
  const testInstance = testRenderer.root
  const parent = testInstance.instance
  const input = testInstance.findByType(Input)
  input.props.onChange(event)
  expect(parent.state.input).toEqual('search term')
})
