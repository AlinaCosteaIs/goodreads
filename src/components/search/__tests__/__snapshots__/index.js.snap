// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component is rendered ok 1`] = `
<div
  className="search"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search field
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-64"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    style={
      Object {
        "margin": "auto 0",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Search
      </span>
    </button>
  </div>
</div>
`;
